tools/uiadm.sh

```diff
@@ -101,56 +127,68 @@ vtar() {
 info() {
 	ui="$INSTALL_PREFIX/bin/ui"
 	if [[ -f "$ui" ]]; then
-		echo "Version: $("$ui" version)"
+		echo "Version: $("$ui" version) $("$ui" stamp)"
+		# shellcheck disable=SC2154
 		echo "URL: https://$Admin_IP:4443"
 	else
 		echo "Not currently installed".
 	fi
 }
 
 get_avail_versions() {
-	vcurl "${URL_PREFIX}/?limit=1000" | \
-	  json -ga name | \
-	  sed -e 's/^smartos\-ui\-//' -e 's/\.tar\.gz$//' | \
-	  sort -t '.' -n -k1,1 -k2,2 -k3,3
-}
-
-avail() {
+	branch="$1"
 	ui="$INSTALL_PREFIX/bin/ui"
 	if [[ -f "$ui" ]]; then
-		version="$("$ui" version)"
-		get_avail_versions | grep -v "$version"
+		stamp="$("$ui" stamp)"
+		marker="&marker=${stamp}"
+		vcurl "${URL_PREFIX}/?limit=1000$marker" | json -ga name |\
+			grep "$branch" | grep -v "$stamp" | grep -v latest
 	else
-		get_avail_versions
+		vcurl "${URL_PREFIX}/?limit=1000" | json -ga name |\
+			grep "$branch" | grep -v latest
 	fi
 }
 
 install() {
+
+	OPTIND=1
+	while getopts "b:" options; do
+	   case $options in
+	      b ) BRANCH="${OPTARG}";;
+	      * ) usage ;;
+	   esac
+	done
+
+	shift $(( OPTIND -1 ))
+
+	# Accept an alternate installation root
+	root="/${2}"
```

Comment:
Deliberately undocumented?  If need be, update usage().

Reply:
Also, do we need a privilege check here or for the whole program?

Reply:
Alternate root is deliberately undocumented. It's intended only for the installer or for testing.

Running it from an alternate location is unsupported (though, it would probably work if you fixed up all the smf paths).

Privilege check is a good idea.

Reply:
Priv checks are on lines 65-70.